woocommerce/woocommerce-android
# Feature/add product categories

Closes #2187 by providing option to add a new product category.

#### Notes
~- This PR is in draft till this [update product categories PR](https://github.com/woocommerce/woocommerce-android/pull/2541) can be merged.~

#### Screenshots
<img src="https://user-images.githubusercontent.com/22608780/84662311-47c04100-af39-11ea-8ea9-f1f51ab16d3a.gif" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/84662319-498a0480-af39-11ea-9573-2d2f95e107f4.gif" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/84662398-66263c80-af39-11ea-86ae-7f516947ae6b.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/84662406-67f00000-af39-11ea-84e4-1292722e3268.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/84662381-60c8f200-af39-11ea-8012-103e82b20c98.png" width="300"/>. <img src="https://user-images.githubusercontent.com/22608780/84662389-6292b580-af39-11ea-8daa-2615e24396d4.png" width="300"/>

#### Testing
- Click on a product from the products tab and click on the categories section. Notice the new `Add Category` button displayed.
  - Add new category without any parent category. Ensure that the new category is added to the site successfully and you are redirected back to the categories list screen with this newly added category already selected.
  - Add new category with parent category. Ensure that the new category is added to the site successfully and you are redirected back to the categories list screen with this newly added category already selected.
  - Add a new product category name and click on the back button. Verify that a discard dialog is displayed. Clicking on `Keep Editing` retains the same screen. Clicking on `Discard`, redirects to the previous screen with no changes.
  - When entering a category name that already exists on the site, an error occurs from the API side. Try entering a duplicate category for a site and verify that an error snackbar is displayed.

##### After adding a new product category
- Verify that the `Done` menu button is visible after adding a new category.
- Click on the back button and verify that a discard dialog is displayed. Clicking on `Keep Editing` retains the same screen. Clicking on `Discard`, redirects to the previous screen with no changes. 
- Click on the `Done` menu button after adding a new product category. Notice the newly added category is visible in the Categories section in the Product Detail screen. 
  - Click on `Update` in the Product Detail screen and verify that the product category is updated successfully

Update release notes:
